include/IReadFile.h

```diff
@@ -43,7 +43,7 @@ namespace io
 
 		//! Get name of file.
 		/** \return File name as zero terminated character string. */
-		virtual const io::path& getFileName() const = 0;
+		virtual const std::string getFileName() const = 0;

```

Comment:
std::filesystem::path

Reply:
also a const reference is required

Reply:
> std::filesystem::path
It used to be `const std::filesystem::path&` but then i found out it is mostly used as a string so later there are plenty of filename.string() calls
> also a const reference is required
i forgot to remove the `const` qualifier but in case i leave string as a return type, i should return std::string not const std::string& cause thats what std::filesystem::path::string() returns

Reply:
in this case and many cases the return type should be `const std::filesystem::path`